oauth2_provider/views/mixins.py

```diff
@@ -74,7 +74,12 @@ def get_server(cls):
         """
         server_class = cls.get_server_class()
         validator_class = cls.get_validator_class()
-        return server_class(validator_class())
+        server_kwargs = (
+            validator_class._get_server_kwargs()
```

Comment:
can we avoid using conditional expression here please?

Reply:
Let me think about it for a moment. I want it to remain compatible with custom validator classes though, not of them necessarily subclass DOT's one.

We could probably just skip the logic I added in that method completely, and just pull the kwargs from settings as is here. But I want to avoid uncertainty as to how configuration values are set and used and custom token generators should be import strings and expires_in stuff already exists as separate variables. Yet ability to pass arbitrary kwargs seems important for custom servers too. Decision to add a validator method for this is based on the idea that we might not be able to figure out exact way to form kwargs for heavily customized setups but if integrators can subclass request validator they can have control over everything that happens there which wouldn't work as easily if we add code to settings module for example (most of the times when using custom server it makes sense to customize validator as well, and server can't be instantiated without a validator).

But I agree that this is kinda ugly and a bit more complex than probably needed. Let me see if I can improve it.

Reply:
I am fine with your addition, but i see here some duplicated code:
 this module L77
 oauth2_provider/oauth2_backends.py L206
 oauth2_provider/oauth2_backends.py L 20

What I think is maybe we can set the validator_class as class attribute or within the `__init__ ', that way we can create a new method `get_server_kwargs`

```Python
def get_server_kwargs(self):
    if hasattr(self.validator_class, '_get_server_kwargs'):
        return self.validator_class._get_server_kwargs()
    else:
        return {}
```

Reply:
Yeah, that would work in that case (I even thought originally to add it to the backend class and not server), but yet another place where a `Server` is currently initialized is `OAuthLibMixin` and I didn't like the idea of relying on `oauthlib_backend_class` being always correctly specified in any its subclasses and be available (it requires a server for instantiation) while request_validator can be assumed to be always present (you can't instantiate a server without it). Note how `get_oauthlib_core` is structured -- it creates a validator instance and a server instance before accessing OAUTH2_BACKEND_CLASS.

Also, two wrapper methods on two different classes just to merge a couple dictionaries essentially feels like an overkill. There ought to be a better way. How about some tweaking in settings? Does making `oauth2_settings.SERVER_KWARGS` a callable (that generates the dictionary based on the other settings) sound any better than this?

Reply:
@MattBlack85 I decided we don't really need a validator method for this too badly. The settings dictionary is enough for most cases and when classes are heavily customized it doesn't matter anyway. So I added a property to the settings class instead. How do you like it now?

Reply:
:+1: :100: